net/yggdrasil/files/yggdrasil.sh

```diff
@@ -97,6 +106,52 @@ proto_yggdrasil_generate_keypair() {
 	public_key=${PrivateKey:64}
 }
 
+proto_yggdrasil_allocate_listen_addresses() {
+	local config="$1"
+
+	# Collect already defined protocols
+	protocols=""
+	_add_address_protocol() {
+		protocols="${protocols}$(echo $1 | cut -d "://" -f1) "
+	}
+	config_list_foreach "$config" listen_address _add_address_protocol
+
+	# Add new address for each previously unspecified protocol
+	for protocol in "tls" "quic"; do
+		if ! echo "$protocols" | grep "$protocol" &>/dev/null; then
```

Comment:
Do not recommend this redirection. 

Advise to: 
```sh
		if ! echo "$protocols" | grep "$protocol" >/dev/null 2>&1 ; then
```

Reply:
Yeah, this could have been even better described simply using `grep -q`.

Btw, what's wrong with `&>...` redirection? `man bash` says it is identical to `2>&1 >...` and imo it looks cleaner.

Reply:
https://www.shellcheck.net/wiki/SC3020

Reply:
Also, what is the `! echo "$protocols"` bit supposed to do? I can guess but I don't like guessing. Syntax is a bit vague. Checking empty return value or string?

Reply:
This whole condition checks if `$protocol` is not present in `$protocols`.
So `!` here acts as a shell keyword that inverting result of the pipeline as a whole (not just `echo`).
And thinking about it, `grep -q --invert-match` would probably fit better here.